### PR TITLE
fix: dropzone behavior

### LINKED ISSUE #965: [bug]: Multiple file uploads seem to trigger twice
### Provide environment information

```bash
System:
    OS: Windows 11 10.0.22631
    CPU: (24) x64 12th Gen Intel(R) Core(TM) i9-12900K
    Memory: 2.90 GB / 31.71 GB
  Binaries:
    Node: 20.16.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.21 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 10.6.0 - C:\Program Files\nodejs\npm.CMD
    pnpm: 9.8.0 - ~\AppData\Roaming\npm\pnpm.CMD
  Browsers:
    Chrome: 128.0.6613.138
    Edge: Chromium (127.0.2651.74)
    Internet Explorer: 11.0.22621.3527
  npmPackages:
    @uploadthing/react: ^7.0.2 => 7.0.2
    typescript: ^5.1.6 => 5.6.2
    uploadthing: ^7.0.2 => 7.0.2
```

### Describe the bug

I followed your remix guide and added a multiple file dropzone
https://github.com/user-attachments/assets/c6cdf72b-9763-41a7-be9a-20bee2032876

### Link to reproduction

https://stackblitz.com/~/github.com/AlemTuzlak/uploadthing-remix

### To reproduce

click on the text of the dropzone and try to upload multiple files
add your key in the .env first though

### Additional information

_No response_

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by juliusmarminge**
closes #965 

thought this was fixed in #886

### Done
- [x] react
  - [x] fixes file dialog opening twice when clicking the input element
- [x] solid
  - [x] fixes file dialog opening twice when clicking the input element 
  - [x] unifies behavior with React components
- [x] vue
  - [x] fixes file dialog opening twice when clicking the input element
  - [x] adds abortion
- [x] svelte button
  - [x] unifies behavior with React components
- [x] Reorganize so that things appear in the same order for each framework to ease maintenance of keeping these in-line for the future
- [x] svelte dropzone

## Release Notes

- **New Features**
  - Introduced a new `Cancel` component for improved user feedback during upload processes.
  - Enhanced `UploadDropzone` and `UploadButton` components with refined state management and event handling for better user experience.

- **Dependency Updates**
  - Removed the `@uploadthing/dropzone` dependency from various packages.
  - Added `file-selector` dependency (version `0.6.0`) to support file selection functionality.

- **Documentation**
  - Updated TypeScript configuration references for improved clarity on Next.js integration.

- **Bug Fixes**
  - Improved error handling and upload progress management, including specific feedback for aborted uploads.
  - Expanded file size limits for various file types in the upload configuration.

These updates streamline file upload processes and enhance the overall functionality of the application.

### CODE REVIEW

File: packages/solid/src/create-uploadthing.ts [hunk 1]
```diff
@@ -117,7 +116,7 @@ export const INTERNAL_createUploadThingGen = <
       }
     };
 
-    const routeConfig = useRouteConfig(initOpts.url, endpoint as string);
+    const routeConfig = createRouteConfig(initOpts.url, endpoint as string);
```
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply packages/solid/src/create-uploadthing.ts:119]: @coderabbitai we use `useX` as a convention for React and Vue whereas solid and svelte have `createX` instead. This is to match built-ins for each library and also ecosystem packages

**Review by t3dotgg — APPROVED** [review]
Looks good, ty for the grind getting all frameworks to behave

File: packages/react/src/components/button.tsx [hunk 2]
```diff
@@ -221,7 +237,7 @@ export function UploadButton<
         return "Loading...";
       }
       case "uploading": {
-        if (uploadProgress === 100) return <Spinner />;
+        if (uploadProgress >= 100) return <Spinner />;
```
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply packages/react/src/components/button.tsx:240]: Haha idk. Math is hard. Probably a bug in here somewhere but felt out of scope so this was a quick fix to get the UI to behave. Will look at the progress calculation later

https://github.com/pingdotgg/uploadthing/blob/main/packages/react/src/useUploadThing.ts#L84-L97
https://github.com/pingdotgg/uploadthing/blob/main/packages/uploadthing/src/internal/upload.browser.ts#L157-L168

**Comment by AlemTuzlak** [comment]
Thanks for the fix!

**Pull request merged**